code/game/objects/items/devices/scanners.dm

```diff
@@ -1006,21 +1006,24 @@ GENE SCANNER
 		symptomholder.Finalize()
 		symptomholder.Refresh()
 		to_chat(user, "<span class='warning'>you begin isolating [chosen].</span>")
-		if(do_mob(user, AM, (600 / scanner.rating)))
-			create_culture(symptomholder, user)
-	else if(do_mob(user, AM, (timer / scanner.rating)))
-		create_culture(A, user)
+		if(do_mob(user, AM, (600 / scanner.rating + 1)))
+			create_culture(symptomholder, user, AM)
+	else if(do_mob(user, AM, (timer / scanner.rating + 1)))
+		create_culture(A, user, AM)
 
 /obj/item/extrapolator/proc/create_culture(var/datum/disease/advance/A, mob/user)
-	if(cooldown > world.time - (1200 / scanner.rating))
+	if(cooldown > world.time - (12 / scanner.rating))
 		to_chat(user, "<span class='warning'>The extrapolator is still recharging!</span>")
 		return FALSE
+	if(!user.get_item_for_held_index(user.active_hand_index))
+		to_chat(user, "<span class='warning'>The extrapolator must be held in your active hand to work!</span>")
+		return FALSE
```

Comment:
Instead of doing this, you should probably replace the `do_mob` with `do_after`.

I believe the way your code is written at the moment, you don't need to be holding the extrapolator in your active hand at all times, only when the action is finishing.

`do_after` has a `needhand` argument, set to `1` by default, that automatically cancels the action if you stop holding what you started with at any time - be it by switching active hands or dropping the item.

Reply:
ah, thats a good point. tommorow, then